src/runners/RunnerUtil.ml

```diff
@@ -57,45 +57,210 @@ let import_lib id =
         plog (sprintf "Successfully imported external library %s\n" name);
         (lmod, initf)
 
+(* An auxiliary data structure that is homomorphic to libtree, but for namespaces.
+   Think of this as a field "namespace" in the "Syntax.libtree". It isn't added
+   to the type itself because we want to eliminate the idea of namespaces right here. *)
+type 'a nspace_tree = 
+  {
+    nspace : 'a ident option;
+    dep_ns : 'a nspace_tree list;
+  }
+
+(* light-weight namespaces. prefix all entries in lib with their namespace. *)
+let eliminate_namespaces lib_tree ns_tree =
+
+  (* Prefix definitions in lib with namespace (and rewrite their uses). 
+     Also, rewrite uses in lib that are in env. This is for names imported by lib. 
+     Returns renamed library and a list of names that are defined in this library. *)
+  let rename_in_library env lib namespace =
+    let (rev_entries, _, def_names) = List.fold lib.lentries ~init:([], env, [])
+    ~f:(fun (accentries, accenv, accnames) entry ->
+      (* check if id is in env and prefix it with a namespace. *)
+      let check_and_prefix_id env id =
+        match List.Assoc.find env ~equal:(=) (get_id id) with
+        | Some ns when ns <> "" ->
+          let nname = ns ^ "." ^ (get_id id) in
+          (* Printf.printf "Adding namespace %s to name %s = %s\n" ns (get_id id) nname; *)
```

Comment:
do we still need this commented out piece of code?

Reply:
I think not, but it could be a debug aid. I'll remove it if you prefer.